src/client/render/pipeline.cpp

```diff
@@ -179,6 +179,21 @@ void TextureBufferOutput::activate(PipelineContext &context)
 
 	// Use legacy call when there's single texture without depth texture
 	// This binds default depth buffer to the FBO
+
+	// This is currently neccessary for a working depth buffer in the following cases:
+	// - post-processing disabled, undersampling enabled
+	//   (addUpscaling specifies no depth texture)
+	// - post-processing disabled, 3d_mode = sidebyside / topbottom / crossview
+	//	 (populateSideBySidePipeline specifies no depth texture)
+	// - post-processing disabled, 3d_mode = interlaced (probably)
+	//   (populateInterlacedPipeline specifies no depth texture)
+	// With post-processing enabled, only a fullscreen rectangle is rendered
+	// to the textures created in the functions listed above, so a depth buffer
+	// is not needed.
+	// With post-processing disabled, the world is actually rendered to these
+	// textures, so a depth buffer is needed.
+	// https://irc.minetest.net/minetest-dev/2022-10-04#i_6021940
```

Comment:
i think that's this code probably:
https://github.com/minetest/minetest/blob/50b75233364c89235ec2f8f77fb54767a228e395/irr/src/CNullDriver.cpp#L575-L579

Reply:
indeed, that's the code that creates the depth texture here

I originally started to investigate this because this piece of code generated warnings for me